etcdserver/api/v3rpc/key.go

```diff
@@ -55,6 +64,7 @@ func (s *kvServer) Put(ctx context.Context, r *pb.PutRequest) (*pb.PutResponse,
 		return nil, togRPCError(err)
 	}
 
+	resp.Header = s.newResponseHeader(resp.Header.Revision)
```

Comment:
i do not quite understand this line. Why do we assign resp.Header.Revision back to resp.Header.Revision?